src/h5web/dataset-visualizer/hooks.ts

```diff
@@ -0,0 +1,35 @@
+import { useContext } from 'react';
+import { VisContext, VisProps } from './VisProvider';
+
+export function useProps(): VisProps {
+  const props = useContext(VisContext);
+
+  if (!props) {
+    throw new Error('Missing Vis provider.');
+  }
+
+  return props;
+}
+
+export function useLineValue(): number[] {
+  const { data, dims } = useProps();
+
+  if (dims.length === 1) {
+    return data;
+  }
+
+  throw new Error('Data not supported by LineVis');
+}
+
+export function useHeatmapValue(): {
+  data: number[][];
+  dims: [number, number];
+} {
+  const { data, dims } = useProps();
+
+  if (dims.length === 2) {
+    return { data, dims: dims as [number, number] };
+  }
+
+  throw new Error('Data not supported by HeatmapVis');
+}
```

Comment:
This is where the mapping will come into play. For now, it is only providing the needed props to the corresponding vis.